datastore/bolt/repo.go

```diff
@@ -132,23 +132,43 @@ func (db *DB) SetRepoKeypair(repo string, keypair *common.Keypair) error {
 
 // DelRepo deletes the repository.
 func (db *DB) DelRepo(repo *common.Repo) error {
-	//TODO(benschumacher) rework this to use BoltDB's txn wrapper
-
-	t, err := db.Begin(true)
-	if err != nil {
-		return err
-	}
 	key := []byte(repo.FullName)
-	err = t.Bucket(bucketRepo).Delete(key)
-	if err != nil {
-		t.Rollback()
+
+	return db.Update(func(t *bolt.Tx) error {
+		err := t.Bucket(bucketRepo).Delete(key)
+		if err != nil {
+			return err
+		}
+		t.Bucket(bucketRepoKeys).Delete(key)
+		t.Bucket(bucketRepoParams).Delete(key)
+
+		// should we just ignore these error conditions? or should
```

Comment:
I think we'll want to handle all errors (even though my initial code did not). I'm cool addressing this now, or coming back in a few weeks and making an effort to cleanup / harden the codebase